common/api/components-react.api.md

```diff
@@ -3249,6 +3367,32 @@ export function useVirtualizedPropertyGridLayoutStorage<T extends Element>(): {
     restore: () => void;
 };
 
+// @beta
+export type Value = NumericValue | InstanceKeyValue | TextValue | BooleanValue | DateValue | EnumValue;
+
+// @beta (undocumented)
+export namespace Value {
+    export function isBoolean(value: Value): value is BooleanValue;
+    export function isDate(value: Value): value is DateValue;
+    export function isEnum(value: Value): value is EnumValue;
+    export function isInstanceKey(value: Value): value is InstanceKeyValue;
+    export function isNumeric(value: Value): value is NumericValue;
+    export function isText(value: Value): value is TextValue;
+}
+
+// @beta
+export interface ValueMetadata {
+    // (undocumented)
+    isNullable?: boolean;
+    // (undocumented)
+    preferredEditor?: string;
+    // (undocumented)
+    type: ValueType;
+}
+
+// @beta
+export type ValueType = "string" | "number" | "bool" | "date" | "dateTime" | "enum" | "instanceKey";
```

Comment:
Similar to `Value`, should we make this extensible?

Reply:
I guess we could make it extensible. One thing I don't like about this approach is that there will be no errors for typos. E.g. 

```ts
function doSomething() {
  const stringValue: ValueMetadata = {
    type: "stirng", // no error for misspelled type
  };

  // no error for misspelled type
  if (stringValue.type === "strng") {
    // handle string
  }
}
```

The intelisense does suggest available string literals so I guess this is not the much of an issue? What do you think?

Reply:
I like this approach over simply adding a comment that new types can be added. What about https://github.com/iTwin/appui/pull/1166#discussion_r1991033472?